messaging/src/androidTest/java/io/lantern/messaging/MessagingTest.kt

```diff
@@ -620,6 +621,120 @@ class MessagingTest : BaseMessagingTest() {
+                                )
+                            }
+                            val eagerAttachment = dog.createAttachment(
+                                eagerPlainTextFile,
+                                "text/plain"
+                            )
+                            val streamAttachment = dog.createAttachment(
+                                "text/plain",
+                                "stream attachment".length.toLong(),
+                                ByteArrayInputStream(
+                                    "stream attachment".toByteArray(
+                                        Charsets.UTF_8
+                                    )
+                                )
+                            )
+                            val imageAttachment = dog.createAttachment(
+                                assetToFile("image.jpg")
+                            )
+                            val audioAttachment = dog.createAttachment(
+                                assetToFile("clap.opus"),
+                            )
+                            assertEquals(
+                                "8.853",
+                                audioAttachment.attachment.metadataMap["duration"],
+                                "audio attachment should include duration metadata"
+                            )
+                            assertEquals(
+                                "application/x-lantern-waveform",
+                                audioAttachment.thumbnail?.attachment?.mimeType,
+                                "audio attachment should include waveform thumbnail"
+                            )
+                            val sentMsg = dog.sendToDirectContact(
+                                dogId,
+                                "hello dog",
+                                attachments = arrayOf(
+                                    badAttachment,
+                                    lazyAttachment,
+                                    eagerAttachment,
+                                    streamAttachment,
+                                    imageAttachment,
+                                    audioAttachment
+                                )
+                            )
+                            val recvMsg = dog.waitFor<Model.StoredMessage>(
+                                sentMsg.dbPath,
+                                "dog should receive message"
+                            )
+
+                            assertEquals(
+                                "hello dog",
+                                recvMsg.text,
+                                "dog should have received correct message text"
+                            )
+                            assertEquals(
+                                6,
+                                recvMsg.attachmentsCount,
+                                "dog should have received 6 attachments"
+                            )
```

Comment:
You need to verify that you can actually read the attachments. If you do, I think you will see that lazy attachments don't work, because they were never encrypted.

Reply:
See https://github.com/getlantern/messaging-android/blob/main/messaging/src/main/java/io/lantern/messaging/CryptoWorker.kt#L204

Reply:
I was getting some exception previously, but that was because Kotlin was trying to download the 'encrypted' attachment, I leave the attachment test just until the upload, but im gonna check it, to see if that can be improved :D